app.js

```diff
@@ -1,52 +1,66 @@
-const bookArray = JSON.parse(localStorage.getItem('bookArray')) || [];
+/* eslint-disable max-classes-per-file */
 
-function updateLocalStorage() {
-  localStorage.setItem('bookArray', JSON.stringify(bookArray));
+class Book {
+  constructor(title, author) {
+    this.title = title;
+    this.author = author;
+  }
 }
 
-function displayBooks() {
-  const bookList = document.getElementById('book-list');
-
-  bookList.innerHTML = '';
-  bookArray.forEach((book, index) => {
-    const div = document.createElement('div');
-    div.setAttribute('id', 'div');
-    bookList.appendChild(div);
-    const titleB = document.createElement('h3');
-    titleB.innerHTML = `${book.book}`;
-    div.appendChild(titleB);
-    const authorB = document.createElement('h3');
-    authorB.innerHTML = `${book.author}`;
-    div.appendChild(authorB);
-    const buttonB = document.createElement('h3');
-    const removeButton = document.createElement('button');
-    removeButton.innerHTML = 'Remove';
-    removeButton.addEventListener('click', () => {
-      buttonB.parentElement.remove();
-      bookArray.splice(index, 1);
-      updateLocalStorage();
+class Library {
+  constructor() {
+    this.bookArray = JSON.parse(localStorage.getItem('bookArray')) || [];
+    this.bookList = document.getElementById('book-list');
+    this.authorInput = document.getElementById('author');
+    this.titleInput = document.getElementById('title');
+    this.addButton = document.getElementById('add');
+
+    this.addButton.addEventListener('click', () => {
+      const author = this.authorInput.value;
+      const title = this.titleInput.value;
+      this.addBook(author, title);
+      this.authorInput.value = '';
+      this.titleInput.value = '';
     });
-    buttonB.appendChild(removeButton);
-    div.appendChild(buttonB);
-    const hr = document.createElement('hr');
-    div.appendChild(hr);
-  });
-}
-function addBook(author, book) {
-  bookArray.push({ author, book });
-  updateLocalStorage();
-  displayBooks();
+
+    this.displayBooks();
+  }
+
+  updateLocalStorage() {
+    localStorage.setItem('bookArray', JSON.stringify(this.bookArray));
+  }
+
+  displayBooks() {
+    this.bookList.innerHTML = '';
+    this.bookArray.forEach((book, index) => {
+      const bookContainer = document.createElement('div');
+      bookContainer.setAttribute('id', 'div');
+      this.bookList.appendChild(bookContainer);
+      const bookDetails = document.createElement('h3');
+      bookDetails.innerHTML = `${book.author} by ${book.title}`;
+      bookContainer.appendChild(bookDetails);
+      const buttonB = document.createElement('h3');
+      const removeButton = document.createElement('button');
+      removeButton.innerHTML = 'Remove';
+
+      removeButton.addEventListener('click', () => {
+        buttonB.parentElement.remove();
+        this.bookArray.splice(index, 1);
+        this.updateLocalStorage();
+      });
```

Comment:
- [ ] There's a glitch in the project where if I add let's say three books and authors and start to delete them randomly, one book persists in localstorage but is deleted from the browser. Upon reload, it appears and when deleted it is then removed from localstorage,  

Before Reload After Deleting Last book
![image](https://user-images.githubusercontent.com/81932478/234414571-3e171420-621a-468b-98e7-dd21992d7b39.png)

After Reloading Page
![image](https://user-images.githubusercontent.com/81932478/234414673-75d021fc-be12-4c25-bfe4-c33daae0aed7.png)
The book comes back








After Deleting It Again
![image](https://user-images.githubusercontent.com/81932478/234414813-26c7b45b-eceb-4bc4-b2a2-8564dc0100ff.png)
Then its deleted now

- [ ] _Kindly fix this issue_